### PR TITLE
publish g4music

### PR DESCRIPTION
**Pull request by neithern**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.neithern.g4music.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ],
+    "modules": [
+        {
+            "name": "g4music",
+            "builddir": true,
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.gnome.org/neithern/g4music"
```
Comment (hfiguiere) [inline com.github.neithern.g4music.json:37]: you need a proper revision

File: com.github.neithern.g4music.desktop.in [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[Desktop Entry]
+Name=G4Music
```
Comment (hfiguiere) [inline com.github.neithern.g4music.desktop.in:2]: this file is never referenced or installed. also `.in` ?

File: com.github.neithern.g4music.appdata.xml.in [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hfiguiere) [inline com.github.neithern.g4music.appdata.xml.in:1]: this file is never referenced or installed. also `.in` ?

And it is way insufficient. missing `release`, `screenshot`, `url` and probably a few other I forget.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
the icon file and the screenshot are never referenced either.

File: com.github.neithern.g4music.json (hunk 1 above)
Comment (neithern) [reply com.github.neithern.g4music.json:55]: updated with version and commit id

File: com.github.neithern.g4music.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "com.github.neithern.g4music.desktop"
+                },
+                {
+                    "type": "file",
+                    "path": "com.github.neithern.g4music.appdata.xml"
+                },
```
Comment (hfiguiere) [inline com.github.neithern.g4music.json:42]: there is a .desktop and a .appdata upstream. So I don't see why you need this here.

**Comment by neithern** [comment]
Updated to follow the app requirements. I'm sorry for my less experience in doing this for the first time

File: com.github.neithern.g4music.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+        <h3 id="toc_1.3">About this project</h3>
+
+        <p>I was inspired by <a href="https://gitlab.gnome.org/World/amberol">Amberol</a> when I started to learn program with GTK4, I found it is so convenient to write code in Vala language: handle signal with closure, async tasks, easy to compile with C code, similar with programming in Kotlin on Android.</p>
+
+    </description>
+        
+    <screenshots>
+        <screenshot type="default">
+            <image>./screen-shot.png</image>
```
Comment (hfiguiere) [inline com.github.neithern.g4music.appdata.xml:43]: this is 100% invalid. this should be a URL.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.neithern.g4music.json [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                {
+                    "type": "file",
+                    "path": "com.github.neithern.g4music.svg"
```
[the remark this answers is not in the document]
Comment (neithern) [reply com.github.neithern.g4music.json:45]: this app icon used in com.github.neithern.g4music.desktop line 6: Icon=com.github.neithern.g4music, should I remove from the json?

File: com.github.neithern.g4music.json (hunk 4 above)
Comment (neithern) [reply com.github.neithern.g4music.json]: removed

File: com.github.neithern.g4music.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (neithern) [reply com.github.neithern.g4music.json]: removed

**Comment by neithern** [comment]
> Build [94893 failed](https://flathub.org/builds/#/builders/32/builds/94893)

the problem caused build failed was fixed

**Comment by hfiguiere** [comment]
> > Build [94893 failed](https://flathub.org/builds/#/builders/32/builds/94893)
> 
> the problem caused build failed was fixed

This is not the right fix. It should be a patch to use the one from upstreanm, and of course a change upstream so that next release don't need the patch.

**Comment by neithern** [comment]
> > > Build [94893 failed](https://flathub.org/builds/#/builders/32/builds/94893)
> > 
> > 
> > the problem caused build failed was fixed
> 
> This is not the right fix. It should be a patch to use the one from upstreanm, and of course a change upstream so that next release don't need the patch.

How to patch? I updated the commit id in the manifest json file, is it the right way?

**Comment by neithern** [comment]
> Build [94920 failed](https://flathub.org/builds/#/builders/32/builds/94920)



> Build [94920 failed](https://flathub.org/builds/#/builders/32/builds/94920)

update the tag name to match the commit id

**Comment by hfiguiere** [comment]
So I will reiterate the request:
Remove the appdata, destkop, icon and screenshot from this repository.


bot, build com.github.neithern.g4music

**Comment by neithern** [comment]
bot, build com.github.neithern.g4music

**Comment by nedrichards** [comment]
bot, build com.github.neithern.g4music

**Comment by neithern** [comment]
> 

done

**Comment by neithern** [comment]
> 

removed

**Comment by hfiguiere** [comment]
bot, build com.github.neithern.g4music

**Comment by neithern** [comment]
> > So I will reiterate the request: Remove the appdata, destkop, icon and screenshot from this repository.
> 
> This sill hasn't been addressed.

I removed these files and Build [95080 successful](https://flathub.org/builds/#/builders/32/builds/95080), what's the next step?

**Comment by neithern** [comment]
@hfiguiere How things going?

**Comment by neithern** [comment]
bot, build com.github.neithern.g4music

**Pull request closed without merging**